jems-testing/src/main/java/org/dmfs/jems/hamcrest/matchers/CharSequenceCharAtMatcher.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.jems.hamcrest.matchers;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+import java.util.Locale;
+
+
+/**
+ * A {@link Matcher} which checks that {@code charAt} of a {@link CharSequence} returns expected values.
+ *
+ * @author Marten Gajda
+ */
+public final class CharSequenceCharAtMatcher extends TypeSafeDiagnosingMatcher<CharSequence>
+{
+    private final String mExpectedValue;
```

Comment:
Why not use `CharSequence` instead of `String` here and in the `CharSequenceMatcher` as well?
Although, I see that we most likely use these with string literal test values.

Reply:
Ah, the `toString` matcher requires a `String` so I went with String everywhere. Anyway I've changed it.